docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/rs-configuration.md

```diff
@@ -182,16 +182,259 @@ The capacity usage on a pool can be monitored using [exporter metrics](../replic
+```text
+kubectl mayastor label node worker-node-1 zone=us-west-1
+kubectl mayastor label node worker-node-2 zone=eu-east-1
+kubectl mayastor label node worker-node-3 zone=us-west-1
+ ```
+
+**Command (Get nodes)**
+ ```text
+kubectl mayastor get nodes -n openebs --show-labels
+ID             GRPC ENDPOINT        STATUS  LABELS
+worker-node-1  65.108.91.181:10124  Online  zone=eu-west-1
+worker-node-3  65.21.4.103:10124    Online  zone=eu-east-1
+worker-node-3  37.27.13.10:10124    Online  zone=us-west-1
+```
+
+### "NodeHasTopologyKey"
+
+The parameter 'NodeHasTopologyKey' will allow the placement of replicas on the node that has label keys that are identical to the keys specified in the storage class.
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  nodeHasTopologykey: |
+    rack
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+Apply the labels on the node using the below command:
+
+**Command**
+```text
+ # kubectl mayastor label node worker-node-1 rack=1
+ # kubectl mayastor label node worker-node-2 rack=2
+ # kubectl mayastor label node worker-node-3 rack=2
+ 
+ # kubectl mayastor get nodes -n openebs --show-labels
+ ID             GRPC ENDPOINT        STATUS  LABELS
+ worker-node-1  65.108.91.181:10124  Online  rack=1
+ worker-node-3  65.21.4.103:10124    Online  rack=2
+ worker-node-3  37.27.13.10:10124    Online  rack=2
+```
+
+In this case, the replica of the volume will be placed on any of the two nodes i.e. 
+- `worker-node-1` and `worker-node-2` or 
+- `worker-node-1` and `worker-node-3` or
+- `worker-node-2` and `worker-node-3`
+as the storage class has `rack` as the value for `nodeHasTopoogyKey` that matches the label key of the node.
+
+### "NodeSpreadTopologyKey"
+
+The parameter 'NodeSpreadTopologyKey' will allow the placement of replicas on the node that has label keys that are identical to the keys specified in the storage class but have different values.
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  nodeSpreadTopologyKey: |
+    zone
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+Apply the labels to the nodes using the below command:
+
+**Command**
+```text
+kubectl mayastor label node worker-node-1 zone=us-west-1
+kubectl mayastor label node worker-node-2 zone=eu-east-1
+kubectl mayastor label node worker-node-3 zone=us-north-1
+ ```
+
+**Command (Get nodes)**
+ ```text
+kubectl mayastor get nodes -n openebs --show-labels
+ID             GRPC ENDPOINT        STATUS  LABELS
+worker-node-1  65.108.91.181:10124  Online  zone=eu-west-1
+worker-node-3  65.21.4.103:10124    Online  zone=eu-east-1
+worker-node-3  37.27.13.10:10124    Online  zone=us-west-1
```

Comment:
This example doesn't show the right intent. Have 2 nodes with the same label, then it will ensure that only one of those 2 nodes will be selected along-with the 3rd node.

Reply:
@sinhaashish Can you kindly look into this.

Reply:
@balaharish7  
Replace  `kubectl mayastor label node worker-node-3 zone=us-north-1` with kubectl `mayastor label node worker-node-3 zone=us-west-1`